tasks/gcp/ReadBackedPhasingTasks.wdl

```diff
@@ -0,0 +1,102 @@
+version 1.0
+
+import "../../structs/gcp/RunTimeSettings.wdl"
+import "../../structs/ReferenceSequence.wdl"
+
+task SelectVariants {
+
+  input {
+    File sample_zarr
+    File called_sites_zarr
+    File phased_sites_zarr
+    String output_basename
+    String contig
+    ReferenceSequence reference
+    RunTimeSettings runTimeSettings
+  }
+
+  Int mem_size = ceil(size(sample_zarr, "GiB") * 3)
+  Int disk_size = ceil(size(sample_zarr, "GiB") * 2)
+
+  command {
+    python /tools/sample_select_variants.py \
+      --sample-genotypes ~{sample_zarr} \
+      --sites-called ~{called_sites_zarr} \
+      --sites-selected ~{phased_sites_zarr} \
+      --output ~{output_basename}.subset.vcf \
+      --contig ~{contig} \
+      --progress
+  }
+
+  runtime {
+    docker: runTimeSettings.select_variants_docker
+    preemptible: runTimeSettings.preemptible_tries
+    cpu: "1"
+    memory: "3.75 GiB"
+    disks: "local-disk 100 HDD"
+  }
+
+  output {
+    File subset_vcf = "~{output_basename}.subset.vcf"
+  }
+}
+    #File subset_vcf_index = "~{output_basename}.subset.vcf.gz.tbi"
+
+
+# TODO:  --max-coverage
+task WhatsHapPhase {
+  input {
+    File input_bam
+    File input_bam_index
+    File subset_vcf
+    String output_basename
+    ReferenceSequence reference
+    RunTimeSettings runTimeSettings
+  }
+
+  command {
+    whatshap phase \
+      -o ~{output_basename}.phased.vcf \
+      --reference ~{reference.ref_fasta} \
+      ~{subset_vcf} ~{input_bam}
+  }
+
+  runtime {
+    docker: runTimeSettings.whatshap_docker
+    preemptible: runTimeSettings.preemptible_tries
+    cpu: "4"
+    memory: "30 GiB"
```

Comment:
This seems high?

Reply:
monitoring script showed it used 11 Gigs of memory for one of the validation samples, running on contig 3L.  So maybe a little high.

Reply:
We can reduce that, maybe 15?

Reply:
I just tried cutting it back to 15, but it failed.  Looking closer at the monitoring log, it used as much as (almost) 18 gig - for both the small and the validation sample.  So, I'm leaving it at 30.